src/main/java/br/com/caelum/viagens/voos/controller/dto/input/NewPassagensInputDto.java

```diff
@@ -0,0 +1,47 @@
+package br.com.caelum.viagens.voos.controller.dto.input;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+
+import javax.validation.constraints.Future;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonFormat.Shape;
+
+import br.com.caelum.viagens.voos.model.Passagem;
+import br.com.caelum.viagens.voos.model.Voo;
+
+public class NewPassagensInputDto {
+	
+	@JsonFormat(shape = Shape.STRING, pattern = "dd/MM/yyyy kk:mm")
+	@Future
+	@NotNull
+	private LocalDateTime dataEHoraDePartida;
+	
+	@JsonFormat(shape= Shape.NUMBER_FLOAT)
+	@Positive
+	@NotNull
+	private BigDecimal valor;
+	
+	public LocalDateTime getDataEHoraDePartida() {
+		return dataEHoraDePartida;
+	}
+	
+	public void setDataEHoraDePartida(LocalDateTime dataEHoraDePartida) {
+		this.dataEHoraDePartida = dataEHoraDePartida;
+	}
+
+	public void setValor(BigDecimal valor) {
+		this.valor = valor;
+	}
+
+	public Voo toModel(Voo voo) {		
+		voo.getAeronave().getAssentos().forEach(assento -> {
+			voo.getPassagens().add(new Passagem(voo, this.dataEHoraDePartida, this.valor, assento));
```

Comment:
Hmmm.. to pensando nessa modelagem. Pensando no negócio mesmo.. ta bem legal que você ta me fazendo refletir bastante sobre as funcionalidades. Quando você for oferecer uma passagem, você não oferece o lugar do avião ainda né? Uma coisa é comprar uma passagem e a outra é escolher o lugar. Digo, pode deixar assim por enquanto, estou só pensando :).

Reply:
ai, eu não sei mais como fazer!!! rs. você tem razão sobre os lugares, por este motivo criei o modelo da aeronave com os assentos. e quando modelei assim, pensei que facilitaria. mas o que mais me confunde é voo X passagem. na minha cabeça, ao buscar passagens, o usuário busca por voos e, ao final do processo de compra, a passagem se concretiza. por isso questionei anteriormente se não era o voo que deveria ter as informações da data de partida e valor. não sei se isso faz sentido...